### DISCUSSION TITLE
How to create a connection pool using the sys user?

### DISCUSSION THREAD

**Discussion by Stephenxman** [oracle/python-oracledb#70] [Q&A] [answered]
Anyone known how to create a connection pool using the sys user?
tks a lot.

**Comment by cjbj** [comment]
In Thin mode, this works:
```
p = oracledb.create_pool(user='sys', password=pw, mode=oracledb.AUTH_MODE_SYSDBA, dsn="localhost/orclcdb")
```

Thin mode doesn't support 'Bequeath' connections, so you always need to give a connection string.  This also means you need to give the SYS user and actual SYS password.

I can see some things I think are buglets with Thick mode.  I'll pass them onto @anthony-tuininga 

The real question: why do you want to create lots of privileged connections?

**Reply by Stephenxman** [reply]
According to our maintenance policy, sys password is random, so I can only connect to sys through thick mode, I need to do a lot of non-interactive checking work with sys users. In the shell I can easily login to sys user through os authentication, so is there any way in oracledb thick mode?

**Comment by cjbj** [comment, marked as answer]
You can currently use SYSDBA via standalone connections in Thin and Thick modes, and via pooled connections in Thin mode.  In Thin mode you will always need to pass the credential and connection string, because this mode doesn't support Oracle's 'Bequeath' connections. In Thick mode you can pass credentials and connection string, or optionally use OS authentication. The following code works when python-oracledb uses the client libraries in $ORACLE_HOME:

```
import oracledb
oracledb.init_oracle_client()
connection = oracledb.connect(mode=oracledb.AUTH_MODE_SYSDBA)
with connection.cursor() as cursor:
    for r in cursor.execute("select user from dual"):
        print(r)
```
